Support for custom deploy user (#54)

lukasbach opened the pull request:
Adds the features requested in #54 

Let me know if I should add or change anything. I'd be happy if the feature would be merged!

lukasbach commented:
@enriikke let me know if I can do something to get this PR merged. I'm happy to change or add anything you're unhappy with with the current changeset!

enriikke commented on index.ts:
```diff
@@ -64,17 +64,32 @@ async function run(): Promise<void> {
-      cwd: `${workingDir}/public`,
-    })
-    await exec.exec(`git config user.email`, [`${github.context.actor}@users.noreply.github.com`], {
+    await exec.exec(`git config user.name`, [core.getInput('git-config-name') || github.context.actor], {
```
To make it a bit easier to parse, let's split up getting the name into a `const` similar to some of the other options: 

```suggestion
    const gitUserName = core.getInput('git-config-name') || github.context.actor
    await exec.exec(`git config user.name`, [gitUserName], {
```

enriikke commented on index.ts:
```diff
@@ -64,17 +64,32 @@ async function run(): Promise<void> {
     })
+    await exec.exec(
+      `git config user.email`,
+      [core.getInput('git-config-email') || `${github.context.actor}@users.noreply.github.com`],
```
and I think we can do the same for this 😃

enriikke commented on action.yml:
```diff
@@ -27,6 +27,18 @@ inputs:
     required: false
     default: "."
+  commit-message:
+    description: "The commit message used for pushing changes to the deploy branch. Use the identifier {sha} to replace with the commit SHA."
```
I really like the initiative here to allow to still include the `sha` but I would like to introduce this in a simpler form by only allowing a string message without the identifier. We could always introduce that in the future if there's a need. 😃

enriikke commented on index.ts:
```diff
@@ -64,17 +64,32 @@ async function run(): Promise<void> {
-      cwd: `${workingDir}/public`,
-    })
+
+    const commitMessageInput = core.getInput('commit-message')
```
This is great! Let's also include the default message here so we can remove the ternary operation below:

```suggestion
    const commitMessageInput = core.getInput('commit-message') || `deployed via Gatsby Publish Action 🎩 for ${github.context.sha}`
```

enriikke reviewed (commented):
This is fantastic @lukasbach! Thank you so much for taking the time to write up a PR and my apologies for taking long to reply. ✨ ❤️ 

I added some suggestions to your PR and also there are a couple of things I would like to see to get it merged:

- Let's update the README to include the new inputs like we do for the existing inputs.
- It's a bit tricky to write tests for CLI tools like this one but it would be great to include a few to make sure we call `git config ...` with the right thing based not he inputs.

lukasbach commented:
Hi @enriikke. Thanks for your suggestions, I've added them to the changeset. I've documented the inputs in the readme, added test cases for the arguments, removed the sha attribute from the commit message and formatted as you suggested. Feel free to let me know if there is something else I can do.

Also, I had issues using the nullish coalescing operator that you suggested (``??``), because the test case run in Github action complained that that's not valid Typescript. The project definitely uses a Typescript version that supports it, so I'm not sure if that's on jest-ts or something else maybe. Locally I did not get any error. Anyway, I've changed it to ``||`` because the variable can only either be a string or undefined anyways at this point.

enriikke reviewed (approved):
This is great! Thank you @lukasbach ✨

lukasbach commented:
Cool, happy that I could help :)
